rustls/src/client/hs.rs

```diff
@@ -136,6 +136,12 @@ impl InitialState {
     }
 
     fn emit_initial_client_hello(mut self, sess: &mut ClientSessionImpl) -> NextState {
+        // During retries :the client MUST send the same ClientHello without
+        // modification" with only a few exceptions as noted in
+        // https://tools.ietf.org/html/rfc8446#section-4.1.2,
+        // Calculate all inputs to the client hellos that might otherwise
+        // change between the initial and retry hellos here to enforce this.
```

Comment:
If/when this is merged, I will file a follow-up issue to track the remaining work of moving logic from `emit_client_hello_for_retry` to this function.